### PR TITLE
[JVM-packages] Add method to get quantile cuts.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
cc @wbo4958 , does this work for your use case?

File: jvm-packages/xgboost4j/src/native/xgboost4j.cpp [hunk 1]
```diff
@@ -1447,6 +1449,46 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGDMatrixGetStrFea
...
+
+  auto ret = XGDMatrixGetQuantileCut(handle, str_config.c_str(), &str_indptr, &str_data);
+
+  ArrayInterface<1> indptr{StringView{str_indptr}};
+  ArrayInterface<1> data{StringView{str_data}};
+  CHECK_GE(indptr.Shape(0), 2);
+
+  // Cut ptr
+  auto j_indptr_array = jenv->NewLongArray(indptr.Shape(0));
+  std::vector<jlong> j_indptr_tmp(indptr.Shape(0));
```
Comment (trivialfis) [inline jvm-packages/xgboost4j/src/native/xgboost4j.cpp:1475]: We can skip this as it's impossible to overflow.

**Comment by wbo4958** [comment]
Thx. That's Cool, I will integrate this PR into my big PR just in case of resolving much conflicts.

File: jvm-packages/xgboost4j/src/test/java/ml/dmlc/xgboost4j/java/DMatrixTest.java [hunk 2]
```diff
@@ -491,6 +496,29 @@ public void testSetAndGetQueryId() throws XGBoostError {
     int[] qidExpected1 = new int[]{0, 3, 4, 6, 8, 9, 10};
     dmat0.setQueryId(qid1);
     TestCase.assertTrue(Arrays.equals(qidExpected1, dmat0.getGroup()));
+  }
 
+  @Test
+  public void testGetQuantileCuts() throws XGBoostError {
+    DMatrix Xy = this.createDenseMatrix();
```
Comment (wbo4958) [inline jvm-packages/xgboost4j/src/test/java/ml/dmlc/xgboost4j/java/DMatrixTest.java:503]: So any DMatrix has the Quantile Cut?

**Comment by wbo4958** [comment]
I've merged this PR into https://github.com/dmlc/xgboost/pull/10639. Thx

**Pull request closed without merging**